### ISSUE TITLE
Vivawallet problem during checkout payment

### ISSUE THREAD

**Issue by gabrielecettolin** [VivaPayments/API#1479]
I have set up the 'redirect' option for my website but at the moment when a customer pay it return to the checkout and say that there has been a problem with the payment.
Inside my account it comes out this message, is it normal?
![Schermata 2020-03-21 alle 12 18 31](https://user-images.githubusercontent.com/62443074/77225450-6a77cf00-6b6f-11ea-8476-22df97196c9e.png)

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin sorry to hear this. In order for me to investigate further what your issue might be, please send through the following screenshots:

- [plugin name] payment module settings (mask your Merchant ID and API Key).
- Viva Wallet [payment source](https://developer.vivawallet.com/e-commerce-plugins/create-a-payment-source/) settings (include your browser's URL bar in the screenshot).
- Viva Wallet _Websites / Apps_ page (include your browser's URL bar in the screenshot).
- [API Access](https://developer.vivawallet.com/e-commerce-plugins/find-your-merchant-id-and-api-key/) page (mask API Key, include your browser's URL bar in the screenshot).

Thanks!!

**Comment by gabrielecettolin** [comment]
Hi Matthew, thank you in advice for your help!
Here are the screenshots you asked me for:

![Schermata 2020-03-23 alle 11 48 43](https://user-images.githubusercontent.com/62443074/77309397-e0f60780-6cfc-11ea-924f-9c34bee516c2.png)
![Schermata 2020-03-23 alle 11 50 06](https://user-images.githubusercontent.com/62443074/77309407-e2bfcb00-6cfc-11ea-8f02-680ab8c24674.png)
![Schermata 2020-03-23 alle 11 50 57](https://user-images.githubusercontent.com/62443074/77309410-e3f0f800-6cfc-11ea-860a-c87c4b445f30.png)
![Schermata 2020-03-23 alle 11 51 41](https://user-images.githubusercontent.com/62443074/77309412-e4898e80-6cfc-11ea-95ee-dc82d8b6705d.png)

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin in the payment source settings (second screenshot) you need to specify HTTPS rather than HTTP. I hope that will fix things.

**Comment by gabrielecettolin** [comment]
Yes before I used to check HTTPS but it didn't work so later I checked HTTP but it doesn't work anyway... 
The setup seems to be correct

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin just need to check, did you download our official version of the Viva Wallet WooCommerce plugin from the following page?

https://developer.vivawallet.com/e-commerce-plugins/woocommerce/

**Comment by gabrielecettolin** [comment]
Hi Matthew, yes I downloaded the official plugin from your official page.

**Comment by matthewrgourd** [comment]
Ok, can you tell me the exact error message received after the attempted payment? I just tried ordering something via your website as a test but the payment screen never came up. Think I chose the wrong option ((sorry, not an Italian speaker!)

**Comment by gabrielecettolin** [comment]
Hi Matthew, sorry but I disabled the credit card payment option because of the error during the payment. I will now reactivate it so you can try a purchase

**Comment by gabrielecettolin** [comment]
It is now activated

**Comment by gabrielecettolin** [comment]
It is now active

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin I've tried several times but I don't think it's active now. Don't worry, if you could just let me know if there's a specific error message after you try buying something? Or is it literally that it never gets to a success or failure page?

**Comment by gabrielecettolin** [comment]
![WhatsApp Image 2020-03-20 at 18 27 40](https://user-images.githubusercontent.com/62443074/77524562-d74cdb00-6e87-11ea-9571-156e80e11bf6.jpeg)
Hi Matthew, this is the message that appears when someone try to pay with Viva Wallet. If the integration is all correct what could be the problem with it? Thanks.

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin, I've checked all your settings at the Viva Wallet end and they're fine. I don't know if you every enabled our Viva Wallet plugin correctly via WooCommerce but can you try again? I will then again attempt a payment with a card with no funds. There should clearly be shown on your checkout page an option for Viva Wallet when enabled. Thanks!

**Comment by gabrielecettolin** [comment]
Ok I've just activated the payment method if you want to try.

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin thanks. I just tried with a card with insufficient funds and can confirm the same error message. However, I was taken through 3DS and that worked correctly, and there is now a failed transaction showing up on your account, so it looks like it's the wording of the error message that is the problem. It should say "declined due to x reason". I did in fact get one of our development team (@therocket-gr) to test the plugin yesterday and he confirms that a card with sufficient funds and authenticated ok via the issuer will make a payment correctly. I don't know what we're doing about the error message but I will look into it further. In the meantime I think you're fine to enable it but monitor sales coming in closely and keep me posted on customer feedback. Thanks!

**Comment by gabrielecettolin** [comment]
Ok, because I've tried with a card with funds and ok but display always the error message. Also other clients had tried to purchase with viva wallet but every time display the error message and in the backend shows me the order but says "waiting for payment". 
I'm waiting to hear from you for other clarification about the error message. 
Thank you!

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin, sorry to hear, I'm raising with our development team.

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin, sorry for the delay getting back to you. Our development team just let me know that we had a temporary outage which could have affected payments, hence why you may have experienced a problem. Can you let me know if the original issue is still there?

**Comment by gabrielecettolin** [comment]
Hi Matthew, thank you for the clarification, now the plugin seems to process the order but then it doesn't go to the thank you page but instead shows me this page.
![Schermata 2020-03-29 alle 13 31 43](https://user-images.githubusercontent.com/62443074/77848060-3a10e000-71c2-11ea-92da-c10a4103a248.png)

**Comment by gabrielecettolin** [comment]
Also in my Vivawallet Dashboard appear the transaction but in the available balance remain 0.00€. There is a problem with this or we have to wait for a certain time before it is credit?
![Schermata 2020-03-29 alle 13 38 34](https://user-images.githubusercontent.com/62443074/77848103-9a078680-71c2-11ea-956b-8ce6d3ae18e1.png)

**Comment by gabrielecettolin** [comment]
???

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin I notice that you've changed your payment source settings to as below:

![Screenshot 2020-04-01 at 09 44 29](https://user-images.githubusercontent.com/44943019/78117407-8c8a1080-73fd-11ea-832d-3a25c4b96492.png)

This is incorrect. The Viva Wallet for WooCommerce plugin settings should be:

Success URL – index.php?wc-api=wc_vivawallet&vivawallet=success
Failure URL – index.php?wc-api=wc_vivawallet&vivawallet=fail

as mentioned in our [installation guide](https://developer.vivawallet.com/e-commerce-plugins/woocommerce/) at Step 5.

**Comment by gabrielecettolin** [comment]
Hi Matthew, I have set the correct one. The right payment source is the one with the code 2224. Several clients have noticed this problem after the payment was sent. After the payment, they do not redirect at the thank you page with the report of the order.

I'll attach you some screenshot of the problem
![WhatsApp Image 2020-04-01 at 19 51 47](https://user-images.githubusercontent.com/62443074/78172127-d30b5980-7455-11ea-8f70-c6400518558f.jpeg)
![WhatsApp Image 2020-04-01 at 19 50 52](https://user-images.githubusercontent.com/62443074/78172133-d69ee080-7455-11ea-8710-f8c19909c07b.jpeg)

**Comment by gabrielecettolin** [comment]
The order arrives correctly to our backend and also both the two emails (one from the store and from Vivawallet) arrive at the client. But they can't see the report page (thank you page). If it is possible to resolve this issue it would be great!
Thank you in advice for the support.

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin the screenshot shows a problem with the 3DS popup window opening on a smartphone browser. Once the checkout process reaches this point it's not handled by Viva Wallet, but by the card issuer. So I can confirm no issue at our end. You can maybe get them to try in a desktop browser instead?

**Comment by gabrielecettolin** [comment]
Yes, I understand but I have just the same issue, a client has purchased two times in a row because it was not sure of the payment was complete. I can't understand why it would not redirect the customer to the review page.

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin there's no way that could have happened if the 3DS popup appeared as in the screenshot. 3DS authentication (and therefore the purchase) simply wouldn't have completed the first time around.

**Comment by matthewrgourd** [comment]
@gabrielecettolin I just checked the account and I can see what you're referring to. There were two transactions today from the same person for the same amount, one of which will need to be refunded. I suggest this is an isolated incident. However, it is 3DS related rather than linked to our plugin.

**Comment by gabrielecettolin** [comment]
Yes i would proceed to the refund, but the problem occurred because after the payment the customers will not be redirected in the thank you page and he assumes that the payment is not processed. That's the problem.

**Comment by gabrielecettolin** [comment]
Hi Matthew sorry for the insistence but the customers also today have complained about their purchase because they are not sure if it was done or not due to the redirection to a white page. How can I redirect the customer to the thank you page after the payment?

We are now increasing the volume of our store and this is not acceptable every time explaining the customers that is a momentary problem. If this problem can't be fixed I'm assuming to change to a different provider.


Thank you in advice Matthew.

**Comment by matthewrgourd** [comment]
@gabrielecettolin understood, will get back to again when I have more answers from our development team.

**Comment by gabrielecettolin** [comment]
It continues to show a blank page after the payment process. I'm every day in trouble with customers who don't understand if they have paid or not due to this problem. Is someone available to help me with this issue?

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin I've looked into this for you and we can confirm that our plugin is working correctly. You will find this problem coming up from time to time with certain customers. The 3DS is handled by individual card issuers as is the popup. It could also be a localised problem with you customers' devices or browsers. I'm closing this issue as it is something that will always come up in a very small number of cases. Thanks!

**Comment by matthewrgourd** [comment]
Hi @gabrielecettolin I gather you've called one of my colleagues. I have closed this issue because I believe it to be fully investigated. I personally tested multiple times on mobile and everything is working fine. It is going to be just a few customers who have an issue but that is to be expected as 3DS is handled in different ways depending on the issuer. If the redirect has worked and the customer has reached the 3DS screen that means the Viva Wallet part of the process has worked correctly. There is nothing wrong at our end. I have checked your account multiple times and around 95% of the transactions are successful. Sorry if you're not happy with my response but it is correct based on my research into your setup. Thanks and have a good day :-)
